ci-operator/step-registry/baremetalds/sno/hostname/baremetalds-sno-hostname-commands.sh

```diff
@@ -41,29 +56,38 @@ wait_approve_csr() {
   oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' | xargs oc adm certificate approve
 }
 
+wait_for_api
+
 if [[ "$(hostname)" != "another-hostname" ]]
 then
+  echo "Deleting node object before changing the hostname..."
+  oc delete node "$(oc get nodes -ojsonpath='{.items[?(@.metadata.name == "'"$(hostname)"'")].metadata.name}')"
+
   systemctl stop kubelet.service
   # Forcefully remove all pods rather than just stop them, because a different hostname
   # requires new pods to be created by kubelet.
-  crictl rmp --force --all
+  until crictl rmp --force --all &> /dev/null
+  do
+    sleep 2
+  done
   systemctl stop crio.service
+
+  # manually remove multus and OVN client certs, as they are not reconciled after the hostname
+  # change and they contain the hostname in their CN
+  rm -rf /etc/cni/multus/certs/multus-client-*.pem \
+    /var/lib/ovn-ic/etc/ovnkube-node-certs/ovnkube-client-*.pem
+
   hostnamectl hostname another-hostname
-  cd /var/lib/kubelet && rm -rfv !\(config.json\)
+
+  cd /var/lib/kubelet
+  awk '$2 ~ path {print $2}' path=/var/lib/kubelet /proc/mounts |xargs --no-run-if-empty umount
```

Comment:
This ensures that all pod mounts are successfully umount-ed before trying to remove all files under `/var/lib/kubelet` (except `/var/lib/kubelet/config.json`), otherwise the rm command fails.

Reply:
For more context, we started by following [this](https://access.redhat.com/solutions/5152031) and then the node team suggested that we remove all files under `/var/lib/kubelet/` (except for the `config.json`).

Reply:
We will refer to the node team, thanks for your time @vrutkovs.

Reply:
> This ensures that all pod mounts are successfully umount-ed before trying to remove all files under `/var/lib/kubelet` (except `/var/lib/kubelet/config.json`), otherwise the rm command fails.

can you please add a comment to that effect (your explanation seems valuable for those needing to debug)? thx